examples/stable_diffusion_xl/gm/models/trainer_factory_new.py

```diff
@@ -0,0 +1,77 @@
+# This file only applies to static graph mode
```

Comment:
Why add `trainer_factory_new.py`

Reply:
I keep it to encourage the discussion of disentangling the LDMWithLoss and TrainOneStepCell Class so that the development of the two modules will be more independent. Check my previous comment here:
> When I implemented textual inversions, SD-XL did not support training the conditioner. I wrote `train_textual_inversion_new.py` and `gm/models/trainer_factory_new.py` to support training the conditioners, taking `examples/stable_diffusion_v2` as a reference, and wrote the `LatentDiffusionWithLoss` class that can backpropagate gradients through conditioners.
> 
> Later I found Huiyao supported training the conditioner of SD-XL in his recent PR. I wrote `train_textual_inversion.py` based on his PR.
> 
> The reason I kept these two redundant files is to encourage the discussion of disentangling `LatentDiffusionWithLoss` from `TrainOneStepCell` in `gm/models/trainer_factory.py`. I think keeping the two parts independent will be easier for future implementation. `gm/models/trainer_factory_new.py` is a reference file and **will be deleted before merging**.
> 
> As for the training speed of `train_textual_inversion_new.py` (referred to as `new`) and `train_textual_inversion.py` (referred to as `old`), on 910A, `old` was about 50ms faster than `new` for each step (the step time was around 4000ms). On 910B, their time difference was negligible. The convergence speeds of the two scripts were similar and the generated images were similar too.

I totally understand that the disentanglement will be not easy as it affects so many other modules. The code refactoring might be a future work. But I want to discuss the necessity and workload of it with you.

Reply:
The two files have been deleted.